### PR TITLE
Re-enable sending H2 GO_AWAY frame

### LINKED ISSUE #1296: Send GO_AWAY frame when closing an H2 connection
Not currently causing any issues, but wanted to add an issue for tracking. 

Linkerd currently closes a connection without sending a GO_AWAY frame due to a now-resolved netty bug.

https://github.com/linkerd/linkerd/blob/master/finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4H2Writer.scala#L71

### PR DESCRIPTION
**Pull request by hawkw**
As described in #1296, we disabled some code for sending a `GO_AWAY` frame when closing an H2 connection, because of an upstream issue in Netty (netty/netty#5307).

Since that issue was fixed in netty/netty@79f2e3604e4c2f1ff6ed6edf04e660c3f46e0200, we should be able to re-enable this code.

All of our existing tests [pass on CI](https://circleci.com/gh/linkerd/linkerd/5260?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link) with these changes. 

Fixes #1296

### CODE REVIEW

**Comment by hawkw** [comment]
I haven't added any new tests for the `GO_AWAY` frame, but I can, if anyone thinks I ought to – might need some advice on what behavior to expect?

File: finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4H2Writer.scala [hunk 1]
```diff
@@ -58,21 +58,18 @@ private[netty4] trait Netty4H2Writer extends H2Transport.Writer {
...
+    }
+    new DefaultHttp2GoAwayFrame(code)
+  }
 
   override def goAway(err: GoAway, deadline: Time): Future[Unit] = {
-    // XXX Our version of netty has a bug that prevents us from
-    // sending AWAYs, so just close until that's fixed.
-    // See: https://github.com/netty/netty/issues/5307
-    //write(goAwayFrame(err))
+    write(goAwayFrame(err))
```
Comment (adleong) [inline finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4H2Writer.scala]: Should we wait until the GoAway is written before closing?  ie flatMap these together?  I don't really know this part of the spec...

Comment (hawkw) [inline finagle/h2/src/main/scala/com/twitter/finagle/buoyant/h2/netty4/Netty4H2Writer.scala]: @adleong I suspect you're probably right. I wasn't really sure whether the commented-out code here was correct, so I was hoping to get review feedback on it...

**Comment by adleong** [comment]
I think we still need to manually test that the go_away is sent with wireshark, nghttp, and/or netty frame logging.

**Comment by hawkw** [comment]
@adleong okay, sounds good. I'll get on that, then!

**Comment by adleong** [comment]
@hawkw any progress on manually testing this?

**Comment by hawkw** [comment]
@adleong, sorry, I've been so busy with response classification that I completely forgot about this – I'll look into it, thanks for the reminder.

**Pull request closed without merging**